src/pages/blog.xml.ts

```diff
@@ -0,0 +1,34 @@
+// src/pages/rss.xml.js
+import rss from '@astrojs/rss';
+import { getCollection } from 'astro:content';
+import type { APIContext } from 'astro';
+
+// https://docs.astro.build/en/reference/api-reference/#endpoint-context
+export async function GET(context: APIContext) {
+	const posts = await getCollection('docs', ({ id }) => {
+		return id.startsWith('blog');
+	});
+
+	posts.sort((a, b) => {
+		const dateA = a.data.date;
+		const dateB = b.data.date;
+		if (dateA && dateB) {
+			if (dateA < dateB) return 1;
+			if (dateA > dateB) return -1;
+			return 0;
+		} else return 0;
+	});
+
+	return rss({
+		title: 'Tauri Blog',
+		description: 'The cross-platform app building toolkit',
+		// @ts-expect-error
```

Comment:
Why is TypeScript yelling here?

Reply:
oh, good question. It says that "Type 'URL | undefined' is not assignable to type 'string | URL'."
I could throw a `context.site as URL`. What do you think?

Reply:
That should do the work or even a typeguard. W/e you decide works for me 😀